### PR TITLE
Allow Override for Id and ModfiedDate Fields

### PR DESCRIPTION
**Pull request by kchitalia**
@smcnulty-sfdc

### CODE REVIEW

File: SalesforceSDK/Salesforce.SDK.SmartSync/Model/SyncDownTarget.cs [hunk 1]
```diff
@@ -147,5 +152,35 @@ public int GetTotalSize()
...
+
+        public long GetMaxTimeStamp(JArray jArray)
+        {
+            long maxTimeStamp = Unchanged;
+            foreach (JToken t in jArray)
+            {
+                var jObj = t.ToObject<JObject>();
+                if (jObj != null)
+                {
+                    var date = jObj.ExtractValue<DateTime>(Constants.LastModifiedDate);
```
Comment (wmathurin) [inline SalesforceSDK/Salesforce.SDK.SmartSync/Model/SyncDownTarget.cs]: Shouldn't you use GetModificationDate() instead ?

Comment (kchitalia) [inline SalesforceSDK/Salesforce.SDK.SmartSync/Model/SyncDownTarget.cs]: That's right, sorry I missed that.

**Comment by kchitalia** [comment]
@wmathurin @smcnulty-sfdc

File: SalesforceSDK/Salesforce.SDK.SmartSync/Model/SoqlSyncDownTarget.cs [hunk 2]
```diff
@@ -116,9 +116,9 @@ public override JObject AsJson()
         /// </summary>
         /// <param name="soql"></param>
         /// <returns></returns>
-        public static SyncDownTarget TargetForSOQLSyncDown(string soql)
-        {
-            return new SoqlSyncDownTarget(soql);
-        }
+        //public static SyncDownTarget TargetForSOQLSyncDown(string soql)
```
Comment (bhariharan) [inline SalesforceSDK/Salesforce.SDK.SmartSync/Model/SoqlSyncDownTarget.cs]: Why's this commented out? Can we remove it?

**Comment by bhariharan** [comment]
Looks fine.

File: SalesforceSDK/Salesforce.SDK.SmartSync/Model/SoqlSyncDownTarget.cs (hunk 2 above)
Comment (kchitalia) [inline SalesforceSDK/Salesforce.SDK.SmartSync/Model/SoqlSyncDownTarget.cs]: Sure, will remove that.

**Comment by kchitalia** [comment]
@bhariharan Addressed your comment regarding removing the commented code.

**Comment by bhariharan** [comment]
:+1:

**Pull request merged**